async.c

```diff
@@ -107,16 +106,10 @@ static redisAsyncContext *redisAsyncInitialize(redisContext *c) {
     redisAsyncContext *ac;
     dict *channels = NULL, *patterns = NULL;
 
-    channels = dictCreate(&callbackDict,NULL);
-    if (channels == NULL)
-        goto oom;
-
-    patterns = dictCreate(&callbackDict,NULL);
-    if (patterns == NULL)
-        goto oom;
-
-    ac = hi_realloc(c,sizeof(redisAsyncContext));
-    if (ac == NULL)
+    /* Allocate memory at the start - OOM on failure */
+    if (NULL == (channels = dictCreate(&callbackDict,NULL)) ||
+        NULL == (patterns = dictCreate(&callbackDict,NULL)) ||
+        NULL == (ac = hi_realloc(c,sizeof(redisAsyncContext))))
```

Comment:
this is a convention I'm fond of, I can revert, but I like that all allocations are in a block